### DISCUSSION TITLE
Need help editing paperdoll

### DISCUSSION THREAD

**Discussion by sssilentrequiem** [exult/exult#583] [General]
Hi, newbie modder here; I've been fighting with exult studio for the last few hours trying to edit the avatar's paperdoll. Every time I export the shp file to GIMP, edit and then export it back to studio, it seems the file "deprecates" each time. i.e. i start losing colours, spots randomly turn transparent etc.
Just wanted to figure out what i'm doing wrong :/ Thanks for any help you can provide.
![example](https://github.com/user-attachments/assets/9915d662-3af5-40cd-b512-63a0ac8a9b8b)

**Comment by wench** [comment]
My guess is you are changing gimp out of indexed color mode or doing something else  losing/changing palette information and Exult Studio attempts to make a best guess converting rgb into the game's palette and does badly

**Reply by sssilentrequiem** [reply]
I checked and it is in indexed mode. Could the issue be my version of GIMP? i downloaded the latest one.

should also mention i'm seeing this issue with my custom portrait too

**Comment by wench** [comment]
What does latest version mean exactly? Gimp 2.10.38 or 3.0 RC1. If you are using 3.0 I'd suspect the problem lies with that.

**Reply by sssilentrequiem** [reply]
It's the 2.10.38 one

**Comment by DominusExult** [comment]
can you give a step by step description so we can try to reproduce it?

**Reply by KnightCaptainU7** [reply]
In addition to the step-by-step, can you share the file you used in the example? Just the head, not the full paperdoll.

**Reply by sssilentrequiem** [reply]
let me see if i can remember.. i went into exult studio, loaded up my modded game (black gate keyring). i right-clicked the default blonde male under gumps.vga then clicked edit to get it into gimp. i made my edits by pasting the file i wanted for the "head" of the paperdoll. if i remember correctly the file i pasted became a different layer (a "floating" layer) which i cut, resized then combined with the layer holding the paperdoll. this was saved and then imported as frame, i believe, into studio replacing the default male in gumps.vga

here is the file i used for the head. i dont mind that it got so crunchy (i figured it would, and im not great at making custom pixel art). i just wish it didnt destroy everything around it lol.
![customavi](https://github.com/user-attachments/assets/02440ab8-5660-4577-9478-798824151da0)

and here is the file i used for the character portrait, vs its appearance in-game:
![customportrait](https://github.com/user-attachments/assets/b7748ce3-9423-4e03-9b8b-ac85a751b3ed)
![image-1](https://github.com/user-attachments/assets/1e881571-d898-4421-98ec-fbb3d69597ce)

**Reply by KnightCaptainU7** [reply]
Okay, let me think this out, starting with the conversation image.
1 You got the image you want to use.
2 Looks like it is already reasonably sized.
3 First problem is it doesn't appear to be a 256 color image. We'll need to convert it to use the default U7 palette.

I think the best results require three things:
Irfanview from https://www.irfanview.com/
GIMP which you have. I use 2.8 final from https://www.gimp.org/downloads/oldstable/ because it lets me see the colors on the palette easier. (Don't worry about this right now.)
The Ultima VII palette. [Palette_as_text.txt](https://github.com/user-attachments/files/17964046/Palette_as_text.txt)

You can also get the palette file by loading the game in Exult Studio and export palette 0.

Create a new folder with only the image file you want in it.
In Irfanview, open the image file.
Use the File Menu -> Batch Conversion/Rename.
Set Work as to Batch conversion - Rename result files.
Set the Output to PNG format.
Click the Advanced button, and use these settings.
![image](https://github.com/user-attachments/assets/4793d915-1d50-4045-b307-4165e37976db)
Click OK, then click Start Batch.

Then open GIMP, and import the Palette, name it Ultima 7. You may need to right-click in the palette list and select Import to do this.
Open the PNG version of the image.
In the Image menu, go to Mode -> RGB.
Then again to the Image menu -> Mode -> Indexed.
Set these options:
![image](https://github.com/user-attachments/assets/b8558665-7def-41c8-86cc-5356055c4a41)
You can also try the various color dithering options, but it seems like Irfanview does this better than GIMP does.
Go to the File menu -> Export As -> PNG image.

Import the image to Exult Studio, and it should appear much more like you expect.

You may want to check the original image is the proper size for BG portraits, or add a border to make it look like it belongs in the game. If it is too big, go back to your original and resize it before re-running the above steps.

Here's my results using the above steps.
![Irfan_then_Gimp](https://github.com/user-attachments/assets/a8d0a620-8c30-479e-8054-4af154e271d0)

**Reply by KnightCaptainU7** [reply]
The paperdoll head might be better done manually by taking the existing Avatar head and adding the three curls to it, because paperdoll size is significantly smaller than the conversation face.

**Reply by sssilentrequiem** [reply]
thank you so much! this is exactly what i needed. the conversation image is now fixed and, following your steps (i imported the paperdoll "head" to irfanview and did the conversion), i managed to get the paperdoll to look a lot better.... i do have a new issue however
![image](https://github.com/user-attachments/assets/6596de21-e7e5-4f67-b15d-2f4ea053d163)
there seems to be some weird flickering on my avatar's head now? that's what those black spots are; they kinda glow on and off like other glow-y effects in the game. i tried erasing the hair and repainting it in gimp and also making sure to reconvert to indexed (with the correct palette this time). but nothing's worked.
at this point im just happy that the rest of him looks normal but if this could be fixed too, i'd like to do so. thank you again for your help!

**Reply by KnightCaptainU7** [reply]
The colors in the bottom two rows of the palette are special, in that they cycle. This is how the effects on the magic armors and the Magebane work.
![image](https://github.com/user-attachments/assets/7e37895b-394c-4a80-9f9f-e0f4671d20da)

You'll want to make sure you aren't use those in your post-conversion image. I may need to delete those two rows of the palette and retry the process to avoid them.

It would be the same steps as above, but first going to the limited Ultima palette, then going back to RGB and then back to the full Ultima palette. I haven't tested this yet.

**Reply by DominusExult** [reply]
The weird flickering is because the palette entry used in these spots is the glowing one ;)
Just replace these with a much earlier palette entry.

@KnightCaptainU7 is it possible to disallow the last 12 (or so?) entries to prevent normal editing to use these?

**Reply by KnightCaptainU7** [reply]
Yes @DominusExult. The last two rows are 32 colors. Here's the limited palette:

![image](https://github.com/user-attachments/assets/0de418a9-274b-4e8c-b6ec-51289a0a57ed)

[Limited_Palette_as_text.txt](https://github.com/user-attachments/files/17964742/Limited_Palette_as_text.txt)

**Reply by sssilentrequiem** [reply]
i figured that was the issue! the paperdoll is now completely fixed, thanks once again for the help!
